library/core/class.request.php

```diff
@@ -86,6 +86,22 @@ class Gdn_Request implements RequestInterface {
     /** @var array Request data/parameters, either from superglobals or from a custom array of key/value pairs. */
     protected $_RequestArguments;
 
+    /**
+    /*  Holds domain plus port as a string to avoid multiple calculations
+     * https://github.com/vanilla/vanilla/issues/7617
+     *
+     * @var string ex: vanilla.local:8080
+     */
+    public static $host = false;
+
+    /**
+    /*  Holds url() results to avoid recalls vs same path
+     * https://github.com/vanilla/vanilla/issues/7617
+     *
+     * @var array
+     */
+    public static $urls = [];
```

Comment:
I can try to make it private. But I don't see any reason to make setters and getters - because all this optimization is about saving cpu ticks to avoid not necessary confusing calls. This is flat associative array and it can (imho should)  be accessed directly. Adding extra step with setter and getter brings nothing to solution. Another point the way it should be used - only one function/method is involved at the moment (and I don't see why it should change in the future) to access and to set values in this array.

Reply:
I only mentioned adding accessors *as necessary*. Leaving these properties public means we don't have proper encapsulation. The entire application would have access to read or write whatever values to them. This unnecessarily opens the properties up to data corruption and stops us from making changes to how these values are stored in the future. If we wanted to change the formatting of strings or the overall structure of the data array, we couldn't, because other parts of the application (internal or in the open source community) would potentially be relying on things being a certain way. If other parts of the application *did* need access to these properties, you could add accessor methods. Using the data array internally would not necessitate getters/setters.